_docs/software_support_lifecycle.md

```diff
@@ -0,0 +1,80 @@
+
+As we operate entirely on the graces of our community members being willing to voluntarily contribute to our modules, we also need to be respective of their time and resources and not add undue stress in order to support software that upstream creators no longer support.
+
+To account for this, we no longer support any software or operating system after the date that the maintainers have marked said software or operating system as End of Life. However, this does not mean we will immediately remove code that allows for the module to run on End of Life'd software or operating systems, but the community will no longer be expected to provide support to users for that software.
+
+There are exceptions and the full policy is detailed in the next section.
+
+### Support Lifecycle
+
+#### Operating Systems
+An Operating System is no longer considered supported once it reaches an End of Life date determined by the upstream maintainer* of the operating system. To ensure our users have enough time to adjust, we will take the following steps to ensure as seamless of a transition as possible:
+
+1. A deprecation PR will be created no later that 3 months before an Operating System's End of Life date.
+2. Removal of the End of Life version from the metadata.json and tests to be completed by the End of Life date.
+3. Issues relating to the End of Life operating system will no longer be addressed after the End of Life date.
+4. Code allowing a module to be run on the End of Life operating system will be left in the module until one of the following milestones are reached after which time the code will be removed:
+    1. Maintaining the code prevents the module from moving forward.
+    2. 12 Months after the End of Life date.
+5. Pull Requests will continue to be accepted with the following exceptions:
+    1. Accepting such a pull request would be detrimental to the module's long term goals.
+    2. The code has already been removed as per the standards set in section 4.
+
+*maintainers include, but are not limited to, the community, company, or organization that maintains ownership over a specific piece of software or project.
+
+##### Support Table:
+
+| OS Name | Support Begins | End of Life | Notification PR | End of Support | Removal of Code |
+| ------- | -------------- | ----------- | --------------- | -------------- | --------------- |
+| CentOS/RHEL 6 | July 10th, 2011 | Nov 30th, 2020 | N/A | Nov 30th, 2020 | Nov 30th, 2021 |
+| CentOS/RHEL 7 | July 7th, 2014 | June 30th, 2024 | Mar 31st, 2024 | June 30th, 2024 | June 30th, 2025 |
+| CentOS/RHEL 8 | Sept 24th, 2019 | Dec 31, 2021 | Sept 30th, 2021 | Dec 31, 2021 | Dec 31, 2022 |
+| CentOS Stream | Jan 1st, 2022 | N/A | N/A | N/A | N/A |
```

Comment:
I *think* there is an EOL date as there will be a CentOS 8 Stream and later a CentOS 9 Stream.

Reply:
I believe the CentOS itself only supports the mainline CentOS Stream. CentOS Stream 8 and Stream 9 are for RHEL Dev purposes.

Reply:
```
Each major release will have a branch, similar to how CentOS Linux is currently structured; however, CentOS Stream is designed to focus on RHEL development, so only the latest Stream will have the marketing focus of the CentOS Project.
```

https://centos.org/distro-faq/

Reply:
It is my understanding that while RHEL 8 is its full support cycle, there will be a CentOS 8 stream. During RHEL's full support cycle, new features and hardware enablement are being implemented and those should first land in CentOS 8 stream. It is exactly the purpose of stream to allow anyone a preview of that. The goal here is Continuous Delivery, similar to how modern development should have an always production ready git master branch. There should be CI to cover this, but of course there will be bugs and regressions since it's still software.

After the full support cycle ends, it enters Maintenance Support. In this phase, RHEL should only receive security features and urgent bug fixes. As such, there's a lot less need for integration testing for upcoming releases (essentially what stream allows).

There is a period (roughly 2 years) where development of RHEL 8 and 9 overlap. During this period there should be 2 streams in parallel, but I think marketing focus implies they'll direct users to CentOS Stream 9 rather than 8.

This is based on https://access.redhat.com/support/policy/updates/errata/